src/ctap/data_formats.rs

```diff
@@ -1159,6 +1160,28 @@ impl From<CredentialManagementSubCommandParameters> for cbor::Value {
     }
 }
 
+/// Translates a the partition identifier for an upgrade to A or B.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(test, derive(IntoEnumIterator))]
+#[repr(u32)]
+pub enum UpgradeIdentifier {
+    A = embedded_flash::PARTITION_A_IDENTIFIER,
+    B = embedded_flash::PARTITION_B_IDENTIFIER,
```

Comment:
Why is this the partition addresses? This doesn't look portable across boards. I think we want to use 0 and 1 here and return 0 or 1 in the UpgradeLocations trait. The board defines which location is A or B (this can be deduced from their index in the board).

Reply:
The identifier is now read using the apps own location, comparing that to the partition location, and returning a 0/1 enum.